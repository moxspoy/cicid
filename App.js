import type {Node} from 'react';
import React, {useState} from 'react';
import {
  SafeAreaView,
  ScrollView,
  StatusBar,
  StyleSheet,
  Text,
  TouchableOpacity,
  useColorScheme,
  View,
} from 'react-native';

import {Colors} from 'react-native/Libraries/NewAppScreen';

const App: () => Node = () => {
  const isDarkMode = useColorScheme() === 'dark';
  const [counter, setCounter] = useState(0);

  const backgroundStyle = {
    backgroundColor: isDarkMode ? Colors.darker : Colors.lighter,
  };

  const increment = () => {
    setCounter(counter => setCounter(counter + 1));
  };
  const reset = () => {
    setCounter(0);
  };
  return (
    <SafeAreaView style={backgroundStyle}>
      <StatusBar barStyle={isDarkMode ? 'light-content' : 'dark-content'} />
      <ScrollView
        contentInsetAdjustmentBehavior="automatic"
        style={backgroundStyle}>
        <View
          style={[
            {
              backgroundColor: isDarkMode ? Colors.black : Colors.white,
            },
            styles.title,
          ]}>
          <CustomText size={20}>
            Penghitung Jumlah Pengunjung Vaksin Covid
          </CustomText>
          <CustomText>
            {'saat ini terdapat '}
            <CustomText testID={'counter'}>{`${counter}`}</CustomText>
          </CustomText>
          <Button onPress={increment} text={'TAMBAH'} />
          <Button onPress={reset} text={'RESET'} />
        </View>
      </ScrollView>
    </SafeAreaView>
  );
};

const Button = ({onPress, text}) => (
  <TouchableOpacity onPress={onPress}>
    <View style={styles.button}>
      <CustomText color={Colors.white} center>
        {text}
      </CustomText>
    </View>
  </TouchableOpacity>
);

const CustomText = ({
  children,
  size = 16,
  color = 'black',
  center,
  ...props
}) => (
  <Text
    style={[
      styles.sectionTitle,
      {
        fontSize: size,
        color,
        textAlign: center ? 'center' : 'left',
      },
    ]}
    {...props}>
    {children}
  </Text>
);

const styles = StyleSheet.create({
  sectionContainer: {
    marginTop: 32,
    paddingHorizontal: 24,
  },
  sectionTitle: {
    fontSize: 24,
    fontWeight: '600',
  },
  sectionDescription: {
    marginTop: 8,
    fontSize: 18,
    fontWeight: '400',
  },
  highlight: {
    fontWeight: '700',
  },
  button: {
    backgroundColor: '#04ba5b',
    padding: 10,
    borderRadius: 10,
    marginTop: 20,
  },
  title: {
    padding: 24,
    paddingTop: 80,
  },
});

export default App;
