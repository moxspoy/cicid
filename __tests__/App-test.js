import 'react-native';
import React from 'react';
import App from '../App';

import {fireEvent, render, waitFor} from '@testing-library/react-native';

it('Increment counter', async () => {
  const {getByTestId, getByText, queryByTestId, toJSON} = render(<App />);

  const button = getByText('TAMBAH');
  fireEvent.press(button);

  await waitFor(() => expect(queryByTestId('counter')).toBeTruthy());

  expect(getByTestId('counter').props.children).toBe('1');
  expect(toJSON()).toMatchSnapshot();
});

it('Reset counter', async () => {
  const {getByTestId, getByText, queryByTestId, toJSON} = render(<App />);

  const button = getByText('RESET');
  fireEvent.press(button);

  await waitFor(() => expect(queryByTestId('counter')).toBeTruthy());

  expect(getByTestId('counter').props.children).toBe('0');
  expect(toJSON()).toMatchSnapshot();
});
