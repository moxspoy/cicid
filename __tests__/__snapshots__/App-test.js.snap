// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Increment counter 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#F3F3F3",
    }
  }
>
  <RCTScrollView
    contentInsetAdjustmentBehavior="automatic"
    style={
      Object {
        "backgroundColor": "#F3F3F3",
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "backgroundColor": "#FFF",
            "padding": 24,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontSize": 24,
                "fontWeight": "600",
              },
              Object {
                "color": "black",
                "fontSize": 20,
                "textAlign": "left",
              },
            ]
          }
        >
          Penghitung Jumlah Pengunjung Vaksin Covid
        </Text>
        <Text
          style={
            Array [
              Object {
                "fontSize": 24,
                "fontWeight": "600",
              },
              Object {
                "color": "black",
                "fontSize": 16,
                "textAlign": "left",
              },
            ]
          }
        >
          saat ini terdapat
          <Text
            style={
              Array [
                Object {
                  "fontSize": 24,
                  "fontWeight": "600",
                },
                Object {
                  "color": "black",
                  "fontSize": 16,
                  "textAlign": "left",
                },
              ]
            }
            testID="counter"
          >
            1
          </Text>
        </Text>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#04ba5b",
                "borderRadius": 10,
                "marginTop": 20,
                "padding": 10,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontSize": 24,
                    "fontWeight": "600",
                  },
                  Object {
                    "color": "#FFF",
                    "fontSize": 16,
                    "textAlign": "center",
                  },
                ]
              }
            >
              TAMBAH
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#04ba5b",
                "borderRadius": 10,
                "marginTop": 20,
                "padding": 10,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontSize": 24,
                    "fontWeight": "600",
                  },
                  Object {
                    "color": "#FFF",
                    "fontSize": 16,
                    "textAlign": "center",
                  },
                ]
              }
            >
              RESET
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;

exports[`Reset counter 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#F3F3F3",
    }
  }
>
  <RCTScrollView
    contentInsetAdjustmentBehavior="automatic"
    style={
      Object {
        "backgroundColor": "#F3F3F3",
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "backgroundColor": "#FFF",
            "padding": 24,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontSize": 24,
                "fontWeight": "600",
              },
              Object {
                "color": "black",
                "fontSize": 20,
                "textAlign": "left",
              },
            ]
          }
        >
          Penghitung Jumlah Pengunjung Vaksin Covid
        </Text>
        <Text
          style={
            Array [
              Object {
                "fontSize": 24,
                "fontWeight": "600",
              },
              Object {
                "color": "black",
                "fontSize": 16,
                "textAlign": "left",
              },
            ]
          }
        >
          saat ini terdapat
          <Text
            style={
              Array [
                Object {
                  "fontSize": 24,
                  "fontWeight": "600",
                },
                Object {
                  "color": "black",
                  "fontSize": 16,
                  "textAlign": "left",
                },
              ]
            }
            testID="counter"
          >
            0
          </Text>
        </Text>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#04ba5b",
                "borderRadius": 10,
                "marginTop": 20,
                "padding": 10,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontSize": 24,
                    "fontWeight": "600",
                  },
                  Object {
                    "color": "#FFF",
                    "fontSize": 16,
                    "textAlign": "center",
                  },
                ]
              }
            >
              TAMBAH
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#04ba5b",
                "borderRadius": 10,
                "marginTop": 20,
                "padding": 10,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontSize": 24,
                    "fontWeight": "600",
                  },
                  Object {
                    "color": "#FFF",
                    "fontSize": 16,
                    "textAlign": "center",
                  },
                ]
              }
            >
              RESET
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
